flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/connectors/DynamicSinkUtils.java

```diff
@@ -403,6 +418,59 @@ private static RelNode convertDelete(
         }
     }
 
+    private static RelNode convertUpdate(
+            LogicalTableModify tableModify,
+            DynamicTableSink sink,
+            ContextResolvedTable contextResolvedTable,
+            String tableDebugName,
+            DataTypeFactory dataTypeFactory,
+            FlinkTypeFactory typeFactory,
+            List<SinkAbilitySpec> sinkAbilitySpecs) {
+        if (!(sink instanceof SupportsRowLevelUpdate)) {
+            throw new UnsupportedOperationException(
+                    String.format(
+                            "Can't perform update operation of the table %s because the corresponding dynamic table sink has not yet implemented %s.",
+                            tableDebugName, SupportsRowLevelUpdate.class.getName()));
+        }
+        SupportsRowLevelUpdate supportsRowLevelUpdate = (SupportsRowLevelUpdate) sink;
+        ResolvedSchema resolvedSchema = contextResolvedTable.getResolvedSchema();
+        List<Column> updatedColumns = getUpdatedColumns(tableModify, resolvedSchema);
+        SupportsRowLevelUpdate.RowLevelUpdateInfo updateInfo =
+                supportsRowLevelUpdate.applyRowLevelUpdate(
+                        updatedColumns, RowLevelModificationContextUtils.getScanContext());
```

Comment:
nit: add a local variable for `RowLevelModificationContextUtils.getScanContext()`